### PR TITLE
[AIRFLOW-5313] Add params support for awsbatch_operator

### PR DESCRIPTION
**Pull request by jrowen**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

AWS Batch supports [passing parameters to jobs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/batch.html#Batch.Client.submit_job), and this update enables that functionality in the `AWSBatchOperator`.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

`tests/contrib/operators/test_awsbatch_operator.py` has been updated to include `parameters`.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/awsbatch_operator.py
Comment (milton0825) [inline airflow/contrib/operators/awsbatch_operator.py]: Can we make `parameters` an optional args to maintain backward compatibility?

File: airflow/contrib/operators/awsbatch_operator.py
Comment (milton0825) [inline airflow/contrib/operators/awsbatch_operator.py]: Does the client check the types? If so, I would rather let client validate the input and raise error.

**Comment by milton0825** [comment]
Can you make a jira and make the commit start with `[AIRFLOW-<jira-number>]`

File: airflow/contrib/operators/awsbatch_operator.py
Comment (dazza-codes) [inline airflow/contrib/operators/awsbatch_operator.py:128]: - nit: the PR could focus only on the addition of params (remove any manipulation of overrides)
- nit: leave the responsibility for parsing job-submit options (types) to the client (+1 @milton0825)

File: airflow/contrib/operators/awsbatch_operator.py
Comment (dazza-codes) [inline airflow/contrib/operators/awsbatch_operator.py]: nit: `, 'overrides', 'parameters')` will preserve a cleaner changeset and preserve alphabetical order

File: tests/contrib/operators/test_awsbatch_operator.py
Comment (dazza-codes) [inline tests/contrib/operators/test_awsbatch_operator.py:67]: nit: is the order important?  If the 'nit' above is adopted, the test order changes to a smaller change also.

File: tests/contrib/operators/test_awsbatch_operator.py
Comment (dazza-codes) [inline tests/contrib/operators/test_awsbatch_operator.py:107]: Please add tests for parameter content (esp. the actual default value; the class itself uses a default of `None`) and test "bad" parameter values

Comment (dazza-codes) [reply tests/contrib/operators/test_awsbatch_operator.py:107]: Q: should the class init translate `None` into `{}` as an effective default?

Comment (dazza-codes) [reply tests/contrib/operators/test_awsbatch_operator.py:107]: Q: does the client have any default params?  If it does, should the operator totally override all of them or just merge specific values into the client defaults?  How does adding params to the operator change the job-request payload?

**Comment by dazza-codes** [comment]
Please squash commits into one feature-commit

File: tests/contrib/operators/test_awsbatch_operator.py
Comment (jrowen) [reply tests/contrib/operators/test_awsbatch_operator.py:107]: > Please add tests for parameter content (esp. the actual default value; the class itself uses a default of `None`) and test "bad" parameter values

I modeled the testing after the `containerOverrides` but can add some more testing if desired.

> Q: should the class init translate `None` into `{}` as an effective default?

The [`submit_job`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/batch.html#Batch.Client.submit_job) will accept either `None` or `{}`--I'm happy to change the class init to the latter if desired.

> Q: does the client have any default params? If it does, should the operator totally override all of them or just merge specific values into the client defaults? How does adding params to the operator change the job-request payload?

Default parameters would be defined as part of each specific jobdef and any values passed into the operator will [override the defaults](https://docs.aws.amazon.com/batch/latest/userguide/example-job-definitions.html#example-use-parameters).

Comment (jrowen) [reply tests/contrib/operators/test_awsbatch_operator.py:107]: Please let know if you have any additional testing suggestions.  Thanks.

**Review by dazza-codes — APPROVED** [review]
LGTM - if the CI can pass

**Comment by jrowen** [comment]
Unless I'm mistaken, the CI error doesn't appear to be related to this code update.  What's the best way to re-run it?

**Review by dazza-codes — APPROVED** [review]
LGTM - and the CI is passing now.

File: airflow/contrib/operators/awsbatch_operator.py
Comment (feluelle) [inline airflow/contrib/operators/awsbatch_operator.py]:
```suggestion
        
        .. seealso:: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/batch.html#Batch.Client.submit_job
```

File: airflow/contrib/operators/awsbatch_operator.py
Comment (feluelle) [inline airflow/contrib/operators/awsbatch_operator.py]:
```suggestion
        parameters (templated)
```

File: airflow/contrib/operators/awsbatch_operator.py
Comment (feluelle) [reply airflow/contrib/operators/awsbatch_operator.py]: Your link didn't work. You can also use `.. seealso::` - this way your link will have its own [colored box](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#colored-boxes-note-seealso-todo-and-warnings).

**Comment by feluelle** [comment]
Please also fix the conflicts and rebase onto master. Thanks 👍 :)

File: airflow/contrib/operators/awsbatch_operator.py
Comment (jrowen) [reply airflow/contrib/operators/awsbatch_operator.py]: Seems that `flake8` doesn't like the new url.  Is there a way to bypass this error?
```
airflow/contrib/operators/awsbatch_operator.py:62:111: E501 line too long (130 > 110 characters)
airflow/contrib/operators/awsbatch_operator.py:66:111: E501 line too long (130 > 110 characters)
airflow/contrib/operators/awsbatch_operator.py:70:111: E501 line too long (130 > 110 characters)
```

Comment (feluelle) [reply airflow/contrib/operators/awsbatch_operator.py]: I think you can use `\` to break the line so it does not exceed the line limit.

**Comment by jrowen** [comment]
The boto3 url has been fixed and the PR has been rebased on master.  What else needs to be addressed before this can be merged?

**Review by feluelle — APPROVED** [review]

**Pull request merged**
